lib/events/s3sessions/s3stream.go

```diff
@@ -62,48 +63,69 @@ func (h *Handler) CreateUpload(ctx context.Context, sessionID session.ID) (*even
 		return nil, trace.Wrap(awsutils.ConvertS3Error(err), "CreateMultiPartUpload session(%v)", sessionID)
 	}
 
-	h.Infof("Upload for session %v created in %v.", sessionID, time.Since(start))
-	return &events.StreamUpload{SessionID: sessionID, ID: *resp.UploadId}, nil
+	h.WithFields(logrus.Fields{
+		"upload":  aws.StringValue(resp.UploadId),
+		"session": sessionID,
+		"key":     aws.StringValue(resp.Key),
+	}).Infof("Created upload in %v", time.Since(start))
+
+	return &events.StreamUpload{SessionID: sessionID, ID: aws.StringValue(resp.UploadId)}, nil
 }
 
 // UploadPart uploads part
 func (h *Handler) UploadPart(ctx context.Context, upload events.StreamUpload, partNumber int64, partBody io.ReadSeeker) (*events.StreamPart, error) {
-	start := time.Now()
-
 	// This upload exceeded maximum number of supported parts, error now.
 	if partNumber > s3manager.MaxUploadParts {
 		return nil, trace.LimitExceeded(
 			"exceeded total allowed S3 limit MaxUploadParts (%d). Adjust PartSize to fit in this limit", s3manager.MaxUploadParts)
 	}
 
+	start := time.Now()
+	uploadKey := h.path(upload.SessionID)
+	log := h.WithFields(logrus.Fields{
+		"upload":  upload.ID,
+		"session": upload.SessionID,
+		"key":     uploadKey,
+	})
+
 	params := &s3.UploadPartInput{
 		Bucket:     aws.String(h.Bucket),
 		UploadId:   aws.String(upload.ID),
-		Key:        aws.String(h.path(upload.SessionID)),
+		Key:        aws.String(uploadKey),
 		Body:       partBody,
 		PartNumber: aws.Int64(partNumber),
 	}
 
+	log.Debugf("Uploading part %v", partNumber)
 	resp, err := h.client.UploadPartWithContext(ctx, params)
 	if err != nil {
 		return nil, trace.Wrap(awsutils.ConvertS3Error(err),
 			"UploadPart(upload %v) part(%v) session(%v)", upload.ID, partNumber, upload.SessionID)
 	}
 
-	h.Infof("UploadPart(upload %v) part(%v) session(%v) uploaded in %v.", upload.ID, partNumber, upload.SessionID, time.Since(start))
-	return &events.StreamPart{ETag: *resp.ETag, Number: partNumber}, nil
+	log.Infof("Uploaded part %v in %v", partNumber, time.Since(start))
+	return &events.StreamPart{ETag: aws.StringValue(resp.ETag), Number: partNumber}, nil
 }
 
 func (h *Handler) abortUpload(ctx context.Context, upload events.StreamUpload) error {
+	uploadKey := h.path(upload.SessionID)
+	log := h.WithFields(logrus.Fields{
+		"upload":  upload.ID,
+		"session": upload.SessionID,
+		"key":     uploadKey,
+	})
 	req := &s3.AbortMultipartUploadInput{
 		Bucket:   aws.String(h.Bucket),
-		Key:      aws.String(h.path(upload.SessionID)),
+		Key:      aws.String(uploadKey),
 		UploadId: aws.String(upload.ID),
 	}
+	log.Debug("Aborting upload")
```

Comment:
is there any use in adding the upload ID so we know _which_ upload was aborted/completed?